src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseSpanBasedStringConcat.Fixer.cs

```diff
@@ -0,0 +1,174 @@
+                return;
+
+            var codeAction = CodeAction.Create(
+                Resx.UseSpanBasedStringConcatCodeFixTitle,
+                FixConcatOperationChain,
+                Resx.UseSpanBasedStringConcatCodeFixTitle);
+            context.RegisterCodeFix(codeAction, diagnostic);
+
+            return;
+
+            // Local functions
+
+            bool IsAnyNonConditionalSubstringInvocation(IOperation operation)
+            {
+                var value = WalkDownBuiltInImplicitConversionOnConcatOperand(operation);
+                return value is IInvocationOperation invocation && symbols.IsAnySubstringMethod(invocation.TargetMethod);
+            }
+
+            async Task<Document> FixConcatOperationChain(CancellationToken cancellationToken)
+            {
+                RoslynDebug.Assert(roscharConcatMethod is not null);
+
+                var editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                var generator = editor.Generator;
+
+                SyntaxNode stringTypeNameSyntax = generator.TypeExpressionForStaticMemberAccess(symbols.StringType);
+                SyntaxNode concatMemberAccessSyntax = generator.MemberAccessExpression(stringTypeNameSyntax, roscharConcatMethod.Name);
+
+                //  Save leading and trailing trivia so it can be attached to the outside of the string.Concat invocation node.
+                var leadingTrivia = operands.First().Syntax.GetLeadingTrivia();
+                var trailingTrivia = operands.Last().Syntax.GetTrailingTrivia();
+
+                var arguments = ImmutableArray.CreateBuilder<SyntaxNode>(operands.Length);
+                foreach (var operand in operands)
+                    arguments.Add(ConvertOperandToArgument(symbols, generator, operand));
+
+                //  Strip off leading and trailing trivia from first and last operand nodes, respectively, and
+                //  reattach it to the outside of the newly-created string.Concat invocation node.
+                arguments[0] = arguments[0].WithoutLeadingTrivia();
+                arguments[^1] = arguments[^1].WithoutTrailingTrivia();
+                SyntaxNode concatMethodInvocationSyntax = generator.InvocationExpression(concatMemberAccessSyntax, arguments.MoveToImmutable())
+                    .WithLeadingTrivia(leadingTrivia)
+                    .WithTrailingTrivia(trailingTrivia);
+
+                SyntaxNode newRoot = generator.ReplaceNode(root, concatExpressionSyntax, concatMethodInvocationSyntax);
+
+                //  Import 'System' namespace if it's absent.
+                if (!IsSystemNamespaceImported(context.Document.Project, generator.GetNamespaceImports(newRoot)))
+                {
+                    SyntaxNode systemNamespaceImport = generator.NamespaceImportDeclaration(nameof(System));
+                    newRoot = generator.AddNamespaceImports(newRoot, systemNamespaceImport);
+                }
```

Comment:
Would you be able to check if calling `WithAddImportsAnnotation` on `concatMethodInvocationSyntax` eliminate the need of `IsSystemNamespaceImported`?

Reply:
`WithAddImportsAnnotation` only works for syntax nodes that represent types (just tested it).
I did change the VB fixer to use a case-insensitive comparison (good catch).